pkg/discovery/network/generic.go

```diff
@@ -25,32 +32,13 @@ func discoverGenericNetwork(clientSet kubernetes.Interface) (*ClusterNetwork, er
 		NetworkPlugin: "generic",
 	}
 
-	podIPRange, err := findPodIPRangeKubeController(clientSet)
-	if err != nil {
-		return nil, err
-	}
-
-	if podIPRange == "" {
-		podIPRange, err = findPodIPRangeKubeProxy(clientSet)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	if podIPRange != "" {
+	podIPRange, err := findPodIPRange(clientSet)
```

Comment:
As commented elsewhere, I think we should propagate an error.

Reply:
Fixed as suggested.